lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -383,37 +381,57 @@ def main():
     description = module.params['description']
     vpc_id = module.params['vpc_id']
     rules = rules_expand_sources(rules_expand_ports(module.params['rules']))
+    #rules = rules_expand_ports(module.params['rules'])
+    #print rules_expand_ports(module.params['rules'])
     rules_egress = rules_expand_sources(rules_expand_ports(module.params['rules_egress']))
     state = module.params.get('state')
     purge_rules = module.params['purge_rules']
     purge_rules_egress = module.params['purge_rules_egress']
+    profile_name="myprofile"
 
     if state == 'present' and not description:
         module.fail_json(msg='Must provide description when state is present.')
 
     changed = False
 
-    ec2 = ec2_connect(module)
+    #ec2 = ec2_connect(module)
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+    if not region:
+         module.fail_json(msg="The AWS region must be specified as an "
+                              "environment variable or in the AWS credentials "
+                              "profile.")
+    client, ec2 = ec2_utils.boto3_conn(module, conn_type='both', resource='ec2', endpoint=ec2_url, region = region, **aws_connect_params)
+    #ec2 = ec2_connect(module)
+    #client, ec2 = ec2_utils._boto3_conn('both', 'ec2', **module.params)
+    #ec2 = ec2_connect(module)
 
     # find the group if present
+    # find the group if present
     group = None
     groups = {}
+    security_groups = []
 
     try:
-        security_groups = ec2.get_all_security_groups()
-    except BotoServerError as e:
+        response = client.describe_security_groups()
+        if 'SecurityGroups' in response:
+            security_groups = response.get('SecurityGroups')
+    except exceptions.NoCredentialsError as e:
```

Comment:
Client is returned as not none when a valid region is specified and aws key and secret are None. So, describe security groups fails with NoCredentialsError and also the existing integration test.